dashboard/app/models/script.rb

```diff
@@ -880,12 +880,12 @@ def get_script_level_by_chapter(chapter)
     script_levels[chapter - 1] # order is by chapter
   end
 
-  def get_bonus_script_levels(current_stage, current_user)
+  def get_bonus_script_levels(current_stage)
     unless @all_bonus_script_levels
```

Comment:
Maybe I'm missing something, I thought a script is only cached once here 
https://github.com/code-dot-org/code-dot-org/blob/1432eb1ed5b2dc0c876a4625c75d290cf511d793/dashboard/app/models/script.rb#L535-L538

Bonus levels are not cached there, but in the `@all_bonus_script_levels` class instance variable. If you have to re-fetch/re-process bonus level data every time, isn't it easier to not use `@all_bonus_script_levels` anymore?

Reply:
A better solution, keep the `@all_bonus_script_levels` variable but only do level summarization on-demand.

Reply:
@hacodeorg thanks for the suggestion. i had been thinking that our script cache stored the script summaries, but now i see that it's only the instances, and summaries are always done on-demand. so this problem is unique to the bonus levels since we  were caching the summaries locally on the instance.

can you take a look at the update i pushed to `Script.get_bonus_script_levels` to see if that's what you had in mind? i am still caching `@all_bonus_levels`, but only summarizing them on demand as you suggested.

i will close [FND-1555] since it no longer seems relevant.